test/csp.js

```diff
@@ -214,6 +216,24 @@ describe("take", function() {
   });
 });
 
+describe("offer and poll", function() {
+  mocha.it("should succeed if they can complete immediately", function() {
+    var ch = chan(2);
+    assert.equal(offer(ch, 42), true);
+    assert.equal(offer(ch, 43), true);
+    assert.equal(offer(ch, 44), undefined);
+    assert.equal(poll(ch), 42);
+    assert.equal(poll(ch), 43);
+    assert.equal(poll(ch), undefined);
+  });
+
+  mocha.it("should fail if they can't complete immediately", function() {
+    var ch = chan();
+    assert.equal(poll(ch), undefined);
+    assert.equal(offer(ch, 44), undefined);
+  });
+});
+
```

Comment:
We should also have these tests:
- offer against a pending take
- poll against a pending put
- offer to a closed channel
- poll on a closed channel